src/gp_doctest.cpp

```diff
@@ -1133,3 +1170,213 @@ TEST_CASE("NNI Engine: GraftDAG") {
         "Graft DAG NodeId does not match Host DAG NodeId.");
   }
 }
+
+// Adds NNIs to DAG, then resizes and reindexes GPEngine, then checks that the same
+// node and edge bitsets correspond to the same PLVs and branch lengths before and after
+// AddNodePair.
+TEST_CASE("GPEngine: Resize and Reindex GPEngine after AddNodePair") {
+  using NodeIdMap = std::unordered_map<Bitset, SizeVector>;
+  using EdgeIdxMap = std::unordered_map<Bitset, SizeVector>;
+  using NodePLVMap = std::unordered_map<Bitset, DoubleVector>;
+  using EdgeBranchMap = std::unordered_map<Bitset, double>;
+
+  // Build maps from DAG bitsets to associated GPEngine data members.
+  auto BuildMapsFromDAG = [](const GPDAG& dag, const GPEngine& gp_engine) {
+    NodeIdMap node_map;
+    EdgeIdxMap edge_map;
+    NodePLVMap plv_map;
+    EdgeBranchMap branch_map;
+
+    for (const auto& node_subsplit : dag.GetSortedVectorOfNodeBitsets()) {
+      size_t node_id = dag.GetDAGNodeId(node_subsplit);
+      // node_id_map
```

Comment:
`node_map`, and `edge_map` below. But if you're going to have comments like this I'd suggest describing what  each one does. (Dropping would be fine.)

Reply:
Dropped comments.